swarm/api/filesystem.go

```diff
@@ -99,51 +98,46 @@ func (fs *FileSystem) Upload(lpath, index string, toEncrypt bool) (string, error
 
 	cnt := len(list)
 	errors := make([]error, cnt)
-	done := make(chan bool, maxParallelFiles)
-	dcnt := 0
 	awg := &sync.WaitGroup{}
 
 	for i, entry := range list {
-		if i >= dcnt+maxParallelFiles {
```

Comment:
you have removed important I/O throttling code here. either convert to a worker pattern or preserve as is

Reply:
Yay! I will revert it.